satpy/composites/__init__.py

```diff
@@ -750,6 +750,55 @@ def __call__(self, projectables, nonprojectables=None, **info):
         return proj
 
 
+class SingleBandCompositor(CompositeBase):
+    """Basic single-band composite builder.
+
+    This preserves all the attributes of the dataset it is derived from.
+    """
+
+    def __init__(self, name, **kwargs):
+        """Collect custom configuration values."""
+        super(SingleBandCompositor, self).__init__(name, **kwargs)
+
+    def _get_sensors(self, projectables):
+        sensor = set()
+        for projectable in projectables:
+            current_sensor = projectable.attrs.get("sensor", None)
+            if current_sensor:
+                if isinstance(current_sensor, (str, bytes, six.text_type)):
+                    sensor.add(current_sensor)
+                else:
+                    sensor |= current_sensor
+        if len(sensor) == 0:
+            sensor = None
+        elif len(sensor) == 1:
+            sensor = list(sensor)[0]
+        return sensor
+
+    def __call__(self, projectables, nonprojectables=None, **attrs):
+        """Build the composite."""
+        num = len(projectables)
+        if num != 1:
+            raise ValueError("Can't have more than one band in a single-band composite")
+        mode = attrs.get('mode', 'L')
```

Comment:
Does `mode` make sense here? This is for non-image composites technically ("physical quantities").

Reply:
Well it could be a single band image in the end...

Reply:
True, but that's the default. If someone wants to provide the mode then there is no reason for the default and it should be preserved anyway. Plus this would be adding metadata to a compositor that is "preserving" the metadata. Very image-y.

Reply:
It could also be P

Reply:
or do you mean that mode is already set in attrs ?

Reply:
Yes, if the image is an image then it is already set and if not then when this physical quantity gets saved as an image the `L` mode will be used by default.